pkg/hbone/server.go

```diff
@@ -37,18 +51,87 @@ func NewServer() *http.Server {
 	h2Server := &http2.Server{}
 	handler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		if r.Method == http.MethodConnect {
-			if handleConnect(w, r) {
+			if handleFunc(w, r) {
 				return
 			}
 		} else {
-			log.Errorf("non-CONNECT: %v", r.Method)
 			w.WriteHeader(http.StatusMethodNotAllowed)
 		}
 	})
-	hs := &http.Server{
+	return &http.Server{
 		Handler: h2c.NewHandler(handler, h2Server),
 	}
-	return hs
+}
+
+func handleDoubleConnect(w http.ResponseWriter, r *http.Request, tlsConfig *tls.Config) bool {
```

Comment:
What are we going to actually use this for testing for? Trick a client into thinking the "network gateway" is an echo pod running double hbone?

The reason I ask is that there would, in the real world, not be any code doing double HBONE termination at all. Since it would just terminate outer and TCP proxy the inner

Reply:
Yeah this doesn't have a ton of utility outside of testing a/the client. If you don't want to spin up a kind cluster and just want to test the correctness of a double hbone client, the server is pretty useful. It's the only way I was able to write the client, as an example